add ability to get aliases of a chain via admin api

kadir96 opened the pull request:
closes #281

danlaine commented on api/admin/service.go:
```diff
@@ -128,6 +128,29 @@ func (service *Admin) AliasChain(_ *http.Request, args *AliasChainArgs, reply *a
+	Chain string `json:"chain"`
+}
+
+// GetAliasesOfChainArgs are the arguments for calling GetAliasesOfChain
```
Can you please change to:
`// GetChainAliases are the aliases of the given chain` ?

danlaine commented on api/admin/service.go:
```diff
@@ -128,6 +128,29 @@ func (service *Admin) AliasChain(_ *http.Request, args *AliasChainArgs, reply *a
+}
+
+// GetAliasesOfChain returns the all aliases of a chain
+func (service *Admin) GetAliasesOfChain(_ *http.Request, args *GetAliasesOfChainArgs, reply *GetAliasesOfChainReply) error {
```
Can you please rename to `GetChainAliases` ? (Similar for args, responses, etc.)

danlaine commented on api/admin/client_test.go:
```diff
@@ -176,6 +180,34 @@ func TestAliasChain(t *testing.T) {
+
+		reply, err := mockClient.GetAliasesOfChain("chain")
+
+		if err != nil {
```
Can you please replace with:
```
assert.NoError(t,err)
```

danlaine commented on api/admin/client_test.go:
```diff
@@ -176,6 +180,34 @@ func TestAliasChain(t *testing.T) {
+		if err != nil {
+			t.Fatalf("Unexepcted error: %s", err)
+		}
+		if !reflect.DeepEqual(expectedReply, reply) {
```
Can you please replace with:
```
assert.ElementsMatch(t, expectedReply.Aliases, reply.Aliases)
```

danlaine commented:
Other than the few comments, looks good! Thanks!

kadir96 commented:
@danlaine all resolved! While renaming, I found that the method for getting aliases already there on `api/admin/chain.go` 😄 . Removed the file and moved the method to `service.go` fyi.

StephenButtolph commented on api/admin/client.go:
```diff
@@ -70,6 +70,15 @@ func (c *Client) AliasChain(chain, alias string) (bool, error) {
 }
 
+// GetChainAliases ...
+func (c *Client) GetChainAliases(chain string) (*GetChainAliasesReply, error) {
```
To be a little more user friendly + in line with other clients, I think this should return `([]string, error)` rather than `(*GetChainAliasesReply, error)`

StephenButtolph commented on api/admin/service.go:
```diff
@@ -128,6 +130,29 @@ func (service *Admin) AliasChain(_ *http.Request, args *AliasChainArgs, reply *a
+func (service *Admin) GetChainAliases(r *http.Request, args *GetChainAliasesArgs, reply *GetChainAliasesReply) error {
+	service.log.Info("Admin: GetChainAliases called with Chain: %s", args.Chain)
+
+	ID, err := ids.FromString(args.Chain)
```
It looks like it was already like this... but since we are here can we clean up the `ID` variable? The capitalization isn't in line with the rest of the repo. Perhaps we could change the name to `chainID` or `id`?

kadir96 replied:
Golint wants `chainId` to be `chainID` so I renamed as `id`

StephenButtolph commented on api/admin/service.go:
```diff
@@ -8,6 +8,8 @@ import (
 	"io/ioutil"
 	"net/http"
 
+	"github.com/ava-labs/avalanchego/ids"
```
Super nit picky here, but can this import get moved to be with the other unnamed "github.com/ava-labs/avalanchego/..." imports?

kadir96 replied:
Did not notice that one tbh 😄 . Don't know why goimports did not do it..

kadir96 commented:
@StephenButtolph all resolved! Can you please take another look?
